pkg/distributor/distributor_test.go

```diff
@@ -945,6 +946,59 @@ func TestDistributor_QueryStream_ShouldReturnErrorIfMaxChunksPerQueryLimitIsReac
 	assert.Contains(t, err.Error(), "the query hit the max number of chunks limit")
 }
 
+func TestDistributor_QueryStream_ShouldReturnErrorIfMaxSeriesPerQueryLimitIsReached(t *testing.T) {
+	const maxSeriesLimit = 10
+
+	limits := &validation.Limits{}
+	flagext.DefaultValues(limits)
+	limits.MaxSeriesPerQuery = maxSeriesLimit
```

Comment:
You can set this, but it's ignored, because what's used is the limiter in the context.